core/common/src/main/java/alluxio/MasterInquireClient.java

```diff
@@ -105,7 +109,38 @@ public synchronized String getMasterAddress() {
         CommonUtils.sleepMs(LOG, Constants.SECOND_MS);
       }
     } catch (Exception e) {
-      LOG.error("Error getting the master address from zookeeper. Zookeeper address: {}",
+      LOG.error("Error getting the leader master address from zookeeper. Zookeeper address: {}",
+          mZookeeperAddress, e);
+    }
+
+    return null;
+  }
+
+  /**
+   * @return the list of all the master addresses, null if errors occur
+   */
+  public synchronized List<String> getMasterAddresses() {
```

Comment:
it seems you either return an empty list or throw exception here. 
I don't see where it is possible to return `null`?